src/devices/src/virtio/net/device.rs

```diff
@@ -172,18 +150,12 @@ impl Net {
             | 1 << VIRTIO_RING_F_EVENT_IDX;
 
         let mut config_space = ConfigSpace::default();
-        // Enable feature to configure a MAC address
-        // If not enabled, the driver generates a random MAC address.
-        avail_features |= 1 << VIRTIO_NET_F_MAC;
-        let mac_addr = {
-            if let Some(some_mac_adrr) = guest_mac {
-                *some_mac_adrr
-            } else {
-                // When the MAC address is not specified explicitly, generate one and assign it.
-                generate_mac_address()
-            }
-        };
-        config_space.guest_mac.copy_from_slice(mac_addr.get_bytes());
+        if let Some(mac) = guest_mac {
+            config_space.guest_mac = mac;
+            // When this feature isn't available, the driver generates a random MAC address.
+            // Otherwise, it should attempt to read the device MAC address from the config space.
```

Comment:
Can we reword the below comment better. A suggestion would be:
```suggestion
            // Enable feature to configure a MAC address
            // If not enabled, the driver generates an address
```

Reply:
Updated